src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsProjectRestoreInfo.cs

```diff
@@ -18,6 +18,13 @@ public interface IVsProjectRestoreInfo
         /// </summary>
         string BaseIntermediatePath { get; }
 
+        //  TODO: Update project-system to add this property, and figure out how to sync the updates
```

Comment:
As we discussed in the meeting earlier, there are 2 options here:
1. Use current field `BaseIntermediatePath` to specify `MSBuildProjectExtensionsPath` value so that won't change anything in interface but change the meaning of this property.

2. Although we dont have any dictionary at `IVsProjectRestoreInfo` but we do have a dictionary at `IVsProjectRestoreInfo -> IVsTargetFrameworks -> IVsTargetFrameworkInfo -> IVsProjectProperties` which is to provide project properties per TFM. So Project system can add `MSBuildProjectExtensionsPath` at this level and duplicate it for each TFM.

Both these options aren't ideal, but if I've to choose one then I'll vote for approach#1 since option# 2 exposes more vulnerability. 

@rrelyea @nkolev92